File: web/app/types/models/asset-import-job.ts
import type { AssetImportKind } from './asset-catalog'

export type AssetImportStatus =
  | 'queued'
  | 'discovering'
  | 'running'
  | 'succeeded'
  | 'succeeded_with_warnings'
  | 'failed'

export type AssetImportTriggerType = 'full_scan' | 'single_file'
export type AssetImportWorkItemStatus =
  | 'queued'
  | 'running'
  | 'succeeded'
  | 'succeeded_with_warnings'
  | 'failed'

export interface AssetImportRun {
  id: string
  kind: AssetImportKind
  triggerType: AssetImportTriggerType
  status: AssetImportStatus
  requestedSourceKey: string | null
  requestedAt: string
  startedAt: string | null
  discoveryFinishedAt: string | null
  finishedAt: string | null
  discoveredFileCount: number
  completedFileCount: number
  succeededFileCount: number
  warningFileCount: number
  failedFileCount: number
  error: string | null
}

export type AssetImportJob = AssetImportRun

export interface AssetImportWorkItem {
  id: string
  runId: string
  importKind: AssetImportKind
  sourceKey: string
  sourcePath: string
  sourceHash: string | null
  status: AssetImportWorkItemStatus
  attemptCount: number
  createdAt: string
  startedAt: string | null
  finishedAt: string | null
  totalResourceCount: number
  processedResourceCount: number
  skippedResourceCount: number
  warningCount: number
  error: string | null
  unpublishedAt: string | null
}

export interface AssetImportDiagnostic {
  id: number
  runId: string
  workItemId: string | null
  severity: 'warning' | 'error'
  code: string
  stage: string
  sourceKey: string | null
  objectName: string | null
  message: string
  createdAt: string
}

export interface AssetImportPage<T> {
  items: T[]
  total: number
  page: number
  pageSize: number
}

File: server/src/L2.Studio.Context/Entities/AssetImportJob.cs
namespace L2.Studio.Context.Entities;

public sealed class AssetImportJob
{
    public Guid Id { get; set; }
    public required string Kind { get; set; }
    public required string Status { get; set; }
    public required string SourcePath { get; set; }
    public string? SourceHash { get; set; }
    public DateTimeOffset RequestedAt { get; set; }
    public DateTimeOffset? StartedAt { get; set; }
    public DateTimeOffset? FinishedAt { get; set; }
    public int TotalCount { get; set; }
    public int ProcessedCount { get; set; }
    public int SkippedCount { get; set; }
    public required string WarningsJson { get; set; }
    public string? Error { get; set; }
}
